src/docker/image.rs

```diff
@@ -27,6 +27,12 @@ pub struct PossibleImage {
 }
 
 impl PossibleImage {
+    pub fn target_is_specified(&self) -> bool {
+        // Hostnames cannot start with either of these characters,
+        // nor can image names.
+        !(self.name.starts_with(':') || self.name.starts_with('@'))
+    }
+
```

Comment:
I don't think I like this, I think it would be better to actually deserialize differently, e.g represent the data as


```rust
pub struct PossibleImage {
    pub reference: ImageReference,
    pub toolchain: Vec<ImagePlatform>,
}
pub enum ImageReference {
   // partially qualified reference. with or without tag/digest  
   Name(String),
   /// unqualified reference, only a tag or digest
   Identifier(String),
}
```

However, I don't have a very strong opinion about it and it would be possible to fix afterhand.

I do think that the name of this function should be changed. `image_name_is_specified`

Reply:
I agree, I'll address this.